### ISSUE TITLE
Failure to create a pmml file when using Logistic regression 

### ISSUE THREAD

**Issue by TGanor** [jpmml/sklearn2pmml#246]
Hi Villu,

I'm trying to create a pmml file from the sklearn model below. I use very simple script to  create a PMML pipeline. when i use XGBOOST classifier every thing works well however when i use LogisticRegression(multi_class = "ovr", max_iter = 1000) i get the following error :
```
TypeError                                 Traceback (most recent call last)
<ipython-input-45-ff0c1f36a757> in <module>
      5 
      6 out_file = "tesoct_7.pmml"
----> 7 generate_pmml(pipeline, out_file)

~/SageMaker/Tamir/PMML/inter_pipes.py in generate_pmml(pipeline, file_path, output)
    268         print(open(file_path).read())
    269     sub2java(file_path)
--> 270     generate_debug_pmml(file_path)
    271 
    272 

~/SageMaker/Tamir/PMML/inter_pipes.py in generate_debug_pmml(out_file, newFileName)
    323                 'optype': f'{optype}',
    324                 'dataType': f'{dataType}',
--> 325                 'feature': 'transformedValue'
    326             })
    327             ET.SubElement(sEle, 'FieldRef', attrib={'field': fieldName})

TypeError: SubElement() argument 1 must be xml.etree.ElementTree.Element, not None
```

**Here is my Code** 
```
rawData =  pd.read_parquet('example.parquet')

y = rawData['enrich_finding']
X = rawData.drop('enrich_finding',axis=1)

X = rawData[['ki_hfcustomerext_mobileappretail_ach_vset','requestedamountasentered','ki_internalhighfocusacp_mobileappretail_login_vset',\
            "accountcity","actimizeenrichmentconnectioncity",]]

X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.33, random_state=44)

recorder.features = []
recorder.out_df=True

recorder.features = recorder.features + [(
    ["ki_hfcustomerext_mobileappretail_ach_vset"], 
    [
            SubstringTransformer(15, 20),
            ExpressionTransformer("1 if X[0] == ':[]}' else 0."),
            Alias(CastTransformer(int), name="ki_hfcustomerext_mobileappretail_ach_vset_ne_")

    ], {'alias': "ki_hfcustomerext_mobileappretail_ach_vset_ne_"}
)]

pipeline = PMMLPipeline([
	("recorder", recorder),
#     ("classifier",xgb.XGBClassifier())
	("classifier", LogisticRegression(multi_class = "ovr", max_iter = 1000))
])

pipeline.fit(X_train, y_train)
pipeline.verify(X_train.head(20))
pipeline.verification.precision = 1e-6
pipeline.verification.zeroThreshold = 1e-6

out_file = "tesoct_7.pmml"
generate_pmml(pipeline, out_file)
```

When i use classifier",xgb.XGBClassifier())

There is no error 

Could you please tell me what's wrong with Logistic regression ?

Thanks Tamir

**Comment by vruusmann** [comment]
Your script uses a `generate_pmml(pipeline, out_file)`, which (presumably-) first invokes `sklearn2pmml.sklearn2pmml(pipeline, out_file)`, and then post-processes the generated PMML file using a custom `generate_debug_pmml(out_file)`.

The error ("TypeError: SubElement() argument 1 must be xml.etree.ElementTree.Element, not None") happens inside this `generate_debug_pmml(out_file)`. This is your method, not mine, so I can't fix it.

The error message suggests that the debugger is trying to create a new `OutputField` element that would expose the value of some internal field. If the debugger works with XGBoost and does not work with logistic regression, then this internal field is most likely the GBT booster value (the sum of all decision tree predictions, before applying the logit transform to it).

In conclusion, you should contact the original author of the `generate_debug_pmml(out_file)` method, and ask him/her to insert basic model type checking there. For example, the method could check if it's dealing with an `MiningModel` element (ie. XGBoost) or a `RegressionModel` element (ie. logistic regression), and then take different action.

**Comment by vruusmann** [comment]
This transformation:
```python
[
  SubstringTransformer(15, 20),
  ExpressionTransformer("1 if X[0] == ':[]}' else 0."),
  Alias(CastTransformer(int), name="ki_hfcustomerext_mobileappretail_ach_vset_ne_")
]
```

can be simplified to:
```python
[
  SubstringTransformer(15, 20),
  Alias(ExpressionTransformer("1 if X[0] == ':[]}' else 0.", dtype = int), name="ki_hfcustomerext_mobileappretail_ach_vset_ne_")
]
```

The thing to not is that the `sklearn2pmml.preprocessing.ExpressionTransformer` transformation type supports the `dtype` argument. There is no need for a standalone cast step.

Of course, it would be extra awesome if the `ExpressionTransformer` supported inline string indexing syntax. This way we could get rid of the leading substring step as well.

**Comment by TGanor** [comment]
Thanks  for your quick support it is working now
